h/config.py

```diff
@@ -143,6 +143,11 @@ def configure(environ=None, settings=None):
     es_logger = logging.getLogger("elasticsearch")
     es_logger.addFilter(ExceptionFilter((("ReadTimeoutError", "WARNING"),)))
 
+    # Add API version settings
+    # These cannot be overridden by env
+    settings["api.versions"] = ["v1", "v2"]
```

Comment:
Using simple strings (e.g. `"v2"`) makes these real easy to work with in various parts of the code. Can y'all think of any reason that not using semver formatting here could bite us?

Reply:
I think simple strings are probably best. If we've made incompatible API changes, equivalent to when you would increment the major version number in SemVer, then we'll increment the version number here. If we've only added new functionality (would increment the minor version number in SemVer) or made bug fixes (patch number in SemVer) we'll just deploy those and won't make any API version number change. I think this makes sense as I don't think we'd want to version the API (and keep on supporting those previous versions if request by an `Accept` header??) every time we deploy fix a bug or add a new endpoint or optional argument.

I think SemVer is mostly for dependencies (e.g. Python and JavaScript packages) and not for APIs (or for apps like h, bouncer, via, lms and client). (semver.org itself claims that the benefits are to do with making "dependency management" easier).

The one problem that comes to mind is if someone was writing an h client that was going to work with multiple different instances of h out there in the wild. If all those instances of h were on API v3, but they were all on different versions of v3, so some had certain new features and bug fixes and some didn't, it might be difficult for the client to know which features it could use with which instance of h. Maybe this would be discoverable by handling 4xx responses and the like.

Anyway, we're not in a world of multiple instances of h and having to update a patch or minor version number every time we make an API change sounds like a real pain